test/e2e/node/pod_resize.go

```diff
@@ -0,0 +1,1421 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/diff"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"k8s.io/kubernetes/pkg/features"
+	kubecm "k8s.io/kubernetes/pkg/kubelet/cm"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+const (
+	CgroupCPUPeriod string = "/sys/fs/cgroup/cpu/cpu.cfs_period_us"
+	CgroupCPUShares string = "/sys/fs/cgroup/cpu/cpu.shares"
+	CgroupCPUQuota  string = "/sys/fs/cgroup/cpu/cpu.cfs_quota_us"
+	CgroupMemLimit  string = "/sys/fs/cgroup/memory/memory.limit_in_bytes"
+
+	PollInterval time.Duration = 2 * time.Second
+	PollTimeout  time.Duration = 4 * time.Minute
+)
+
+var (
+	featureGatePostAlpha bool
+	podClient            *framework.PodClient
```

Comment:
please don't use a shared variable for the podClient, just fetch it where it's used. It's much harder to follow tests that share variable state.